File: python/BOJ/2XXX/2168.py
"""
https://www.acmicpc.net/problem/2168
2168.타일 위의 대각선
실버1
풀이1.68ms
"""


def gcd(a, b):
    if a < b:
        a, b = b, a
    if b == 0:
        return a
    return gcd(b, a % b)


x, y = map(int, input().split())
print(x + y - gcd(x, y))

File: kotlin/src/main/kotlin/boj/2000/2168.kt
package boj.`2000`

/*
https://www.acmicpc.net/problem/2168
2168.타일 위의 대각선
풀이.132ms
 */
fun main() {
    var (x, y) = readln().split(" ").map { it.toInt() }
    println(x + y - gcd(x, y))
}

fun gcd(a:Int, b:Int):Int {
    if (b == 0) {
        return a
    }

    return gcd(b, a % b)
}